pyhive/sqlalchemy_hive.py

```diff
@@ -517,6 +518,37 @@ def _check_unicode_description(self, connection):
         # We decode everything as UTF-8
         return True
 
+
+from sqlalchemy.ext.compiler import compiles
+from sqlalchemy.sql.expression import Select
+
+
+@compiles(Select)
+def select_star_if_all_columns(select, compiler, **kwargs):
+    froms = select.froms
+    select_columns = list(select.columns.keys())
```

Comment:
this could break if I do something like `select([(table.c.x + 1).label('x')])`, so I think this needs to check that everything is a Column instance.